### DISCUSSION TITLE
Thoughts on per-language tweaks?

### DISCUSSION THREAD

**Discussion by adrian5** [2giosangmitom/nightfall.nvim#55] [Ideas]
In theory, one could tweak individual languages to make them look even nicer. I have some ideas there.

For example `@tag.attribute.html` inherits from `@tag.attribute` but allows custom overrides for HTML.

**Pros**:

* A more polished/differentiated look for various languages

**Cons**:

* Added maintenance burden if tree-sitter grammars change over time
* More work for the theme to be set? Maybe not too relevant given how fast it is when "compiled".

How do you feel about this? Something to consider or better to avoid?

**Comment by 2giosangmitom** [comment]
I think adding difference highlighting could be interesting, and it might not significantly slow down the plugin. If you want to implement this, expand an option for specific language support to the treesitter integration.

The user can be used like this:

```lua
integrations = {
  treesitter = { enabled = true, html = true, markdown = true, context = true } -- and more
}
```

**Reply by adrian5** [reply]
Where would they be placed? I think giving each language their separate file would be cleaner. Right now we have `/lua/nightfall/groups/integrations/treesitter.lua`. It could be e.g. `treesitter-rust.lua` or a nested directory? You decide.

**Reply by 2giosangmitom** [reply]
We should refactor the treesitter highlights, the structure should look like this

```
├── integrations/
├──── init.lua -- main highlight here
├──── html.lua
├──── markdown.lua
├──── context.lua
```

**Reply by 2giosangmitom** [reply]
Let me know if you'd like to implement this, or I can take care of it as soon as possible. I'm just swamped with deadlines right now :)

**Reply by adrian5** [reply]
Sound good! `integrations/treesitter/` you mean, or together with the other plugins?

Don't worry about this now, there's no rush. I can test some tweaks now and move them into the structure you have chosen before doing a PR.

**Reply by 2giosangmitom** [reply]
`integrations/treesitter`, sorry about that mistake
